### PR TITLE
cmake: Move declarations up.

### PR DESCRIPTION
**Pull request by hakonfam**
Move several declarations further up in the file.

This is done to make future modifications simpler,
as more variables and functions are available.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

### CODE REVIEW

**Comment by mped-oticon** [comment]
FYI, this may be superseded by PR https://github.com/zephyrproject-rtos/zephyr/pull/11551.

File: CMakeLists.txt [hunk 1]
```diff
@@ -308,6 +308,20 @@ zephyr_cc_option(-Wpointer-arith)
...
+    set(APP_SMEM_DEP app_smem_linker)
+  endif()
+  if(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT AND CONFIG_APPLICATION_MEMORY)
+    set(ALIGN_SIZING_DEP app_sizing_prebuilt linker_app_sizing_script)
+  endif()
+  if(CONFIG_ARM)
+  set(PRIV_STACK_DEP priv_stacks_prebuilt)
+  endif()
+endif()
+
```
Comment (SebastianBoe) [inline CMakeLists.txt]: Each independent patch needs to be in it's own commit.

Re-ordering things can break things in unpredictable and subtle ways,
but a clean history can help to debug these issues.

Comment (hakonfam) [reply CMakeLists.txt]: Will try to split it up in more commits.

**Review by SebastianBoe — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (hakonfam) [reply CMakeLists.txt]: check it now @SebastianBoe

Comment (SebastianBoe) [reply CMakeLists.txt]: You are missing toup markers, but I guess you know that?

Comment (SebastianBoe) [reply CMakeLists.txt]: LGTM

**Comment by hakonfam** [comment]
recheck

**Comment by carlescufi** [comment]
@hakonfam can you please rebase?

**Comment by carlescufi** [comment]
@nashif can you take a look at this one?

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@hakonfam seems like there's an issue with CI. I assume this is a timeout, but @nashif could confirm this.

**Comment by hakonfam** [comment]
recheck

**Review by nashif — CHANGES_REQUESTED** [review]
The CI failures are real, something is not right with the code reordering, bad rebase?

**Review by nashif — APPROVED** [review]
